src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -63,75 +75,304 @@ private static void loadModel(Model model, CharSource ontologySource) {
      * @return An AIF validator for the specified ontology
      */
     public static ValidateAIF createForDomainOntologySource(CharSource domainOntologySource) {
+        return create(false, false, false, ImmutableSet.of(domainOntologySource));
+    }
+
+    /**
+     * Create an AIF validator for specified domain ontologies and requirements.
+     * @param ldcFlag Whether or not to validate against the LDC ontology
+     * @param programFlag Whether or not to validate against the program working group's ontology
+     * @param nistFlag Whether or not to validate against the NIST requirements
+     * @param domainOntologySources A user-supplied domain ontology
+     * @return An AIF validator for the specified ontologies and requirements
+     */
+    public static ValidateAIF create(boolean ldcFlag, boolean programFlag, boolean nistFlag,
+                                     ImmutableSet<CharSource> domainOntologySources) {
+        if (!ldcFlag && !programFlag && (domainOntologySources == null || domainOntologySources.isEmpty())) {
+            throw new IllegalArgumentException("Must validate against at least one domain ontology.");
+        }
+
         final OntModel model = ModelFactory.createOntologyModel();
         model.addLoadedImport(INTERCHANGE_URI);
         model.addLoadedImport(AIDA_DOMAIN_COMMON_URI);
 
         // Data will always be interpreted in the context of these two ontology files.
-        ImmutableSet<CharSource> models = ImmutableSet.of(
+        final ImmutableSet<CharSource> aidaModels = ImmutableSet.of(
                 Resources.asCharSource(Resources.getResource(INTERCHANGE_RESNAME), Charsets.UTF_8),
-                Resources.asCharSource(Resources.getResource(AIDA_DOMAIN_COMMON_RESNAME), Charsets.UTF_8),
-                domainOntologySource);
+                Resources.asCharSource(Resources.getResource(AIDA_DOMAIN_COMMON_RESNAME), Charsets.UTF_8)
+        );
+
+        final HashSet<CharSource> models = new HashSet<>(aidaModels);
+        if (domainOntologySources != null) {
+            models.addAll(domainOntologySources);
+        }
+        if (ldcFlag) {
+            models.add(Resources.asCharSource(Resources.getResource(LDC_RESNAME), Charsets.UTF_8));
+        }
+        if (programFlag) {
```

Comment:
Yep, while we decided that ldc, program, and --ont were mutually exclusive, I should have made more internal changes to reflect that better.  I made the switches mutually exclusive, but I should also change it so there's no ldcFlag and programFlag in create(), because that suggests that they can both be true, which is invalid.  On the other hand, I don't want API users to have to know where the ontologies are, create CharSources out of them, put them in a set, etc.  So I'm thinking I might have a new createForLDCOntology and createForProgramOntology public methods that take care of some of that for you.  Behind the scenes, they'll all probably call the same method (taking a Set of CharSources.

This fix will be in the next commit.